a4c-brooklyn-plugin/src/main/java/alien4cloud/brooklyn/BrooklynProvider.java

```diff
@@ -382,15 +403,61 @@ public synchronized void getEventsSince(Date date, int maxEvents, IPaaSCallback<
         BrooklynApi brooklynApi = getNewBrooklynApi();
         try {
             Collection<AbstractMonitorEvent> events = Sets.newHashSet();
-            for (ApplicationSummary appSummary : brooklynApi.getApplicationApi().list(null)) {
+            List<ApplicationSummary> appSummaries = brooklynApi.getApplicationApi().list(null);
+			for (ApplicationSummary appSummary : appSummaries) {
                 String appId = appSummary.getId();
                 String deploymentId = String.valueOf(brooklynApi.getEntityConfigApi().get(appId, appId, "tosca.deployment.id", false));
+
                 for (EntitySummary entitySummary : brooklynApi.getEntityApi().getDescendants(appId, appId, null)) {
                     String entityId = entitySummary.getId();
                     List<TaskSummary> taskSummaries = brooklynApi.getEntityApi().listTasks(appId, entityId);
                     addTasksAndDescendants(taskSummaries, events, deploymentId, date, brooklynApi);
                 }
             }
+
+            for(Entry<String, Optional<DeploymentStatus>> deploymentStatus : ImmutableSet.copyOf(deploymentStatuses.entrySet())) {
+                final String entityId = deploymentStatus.getKey();
+                final Optional<PaaSTopologyDeploymentContext> deployment = Iterables.tryFind(knownDeployments.values(), new Predicate<PaaSTopologyDeploymentContext>() {
+                    @Override
+                    public boolean apply(@Nullable PaaSTopologyDeploymentContext input) {
```

Comment:
Can `input` actually be null? Either handle the null case or remove the annotation.